core/pkg/server/stream_init.go

```diff
@@ -84,9 +85,16 @@ func NewGraphQLClient(
 	settings *settings.Settings,
 	peeker *observability.Peeker,
 ) graphql.Client {
+	// TODO: This is used for service account to associate the run with the
+	// correct user. Note that we are using environment variables here, instead
+	// of the settings object. This is because the settings object will populate
+	// both username and email and it might be incosistent with partial environment
+	// variables. We should consider using the settings object here.
```

Comment:
This part of the comment doesn't make sense to me. What are partial environment variables? Is the behavior in Settings correct, or would it be worth fixing it?

Reply:
oh i see. i meant that if we only specify the email without the username as environment variable, the settings object will assign some username, and it seemed to used in other places.

the problem that i'm not sure what is the correct behavior, it seems wrong to me as we set the username based on this logic: https://github.com/wandb/wandb/blob/main/wandb/sdk/wandb_settings.py#L1741-L1747 

I didn't want to remove this logic in case the username is used in a different place in a different way, just to reduce scope. Hoped we could fix it once we are re-doing the client.